Vue/src/App.vue

```diff
@@ -0,0 +1,117 @@
+      :on-appointment-updating='(e) => { handleAppointmentActions(e, e.newData); }'
+      :data-source='data'
+      :views='[{type: "week"}]'
+      :first-day-of-week='0'
+      current-view='week'
+      :current-date='new Date(2020, 10, 25)'
+      :start-day-hour='9'
+      width='100%'
+      height='100%'
+  />
+</template>
+
+<script>
+import 'devextreme/dist/css/dx.common.css';
+import 'devextreme/dist/css/dx.light.css';
+import { DxScheduler } from 'devextreme-vue/scheduler';
+import { DxPopup, DxPosition, DxToolbarItem } from 'devextreme-vue/popup';
+import { rrulestr, RRule } from 'rrule';
+import { defaultData } from './data';
+
+export default {
+  name: 'App',
+  components: {
+    DxScheduler,
+    DxPopup,
+    DxPosition,
+    DxToolbarItem
+  },
+  data() {
+    return {
+      popupVisible: false,
+      data: defaultData,
+      closeButtonOptions: {
+        text: 'Close',
+        onClick: () => {
+          this.popupVisible = false;
+        },
+      },
+    }
+  },
+  methods: {
+    formattedDate(date) {
+      return  new Date(date.getFullYear(), date.getMonth(), date.getDate())
+    },
+    handleAppointmentActions(e, appointmentData) {
+      let startTime = appointmentData.startDate.getTime();
+      const endTime = appointmentData.endDate.getTime();
+      const recurringAppointment = defaultData.filter((appointment) => appointment?.recurrenceRule).map((appointment) => {
+        return {
+          ...appointment,
+          startDate: new Date(appointment.startDate),
+          endDate: new Date(appointment.endDate),
+        };
+      });
+
+      recurringAppointment.find((appointment) => {
+        const recurrenceOptions = rrulestr(appointment.recurrenceRule);
+        const rule = new RRule({
+          freq: recurrenceOptions.options.freq,
+          interval: recurrenceOptions.options.interval,
+          byweekday: recurrenceOptions.options.byweekday,
+          dtstart: appointment?.startDate,
+        })
+        const betweenDate = rule.between(e.component.getStartViewDate(), e.component.getEndViewDate())
+        const appointmentDuration = appointment.endDate.getTime() - appointment.startDate.getTime();
+
+        if (betweenDate.length > 0) {
+          betweenDate.find((date) => {
+            let recurrentStartTime = date.getTime();
+
+            if(appointment.allDay) {
+              recurrentStartTime = recurrentStartTime - 86400000
+              startTime = this.formattedDate(new Date(startTime)).getTime()
+            }
+            const recurrentEndTime = recurrentStartTime + appointmentDuration;
+
+            if (
+                startTime === recurrentStartTime
+                || startTime > recurrentStartTime && startTime < recurrentEndTime
+                || endTime > recurrentStartTime && endTime <= recurrentEndTime
+            ) {
+              e.cancel = true;
+              this.popupVisible = true;
+            }
+          })
+        }
+      })
+    }
```

Comment:
Let's change our algorithm of detecting overlap between a new appointment and existing recurrent appointments:

**1) Let's move logic to separate pure function:**
Take a look at typescript function in the Angular component comment.
Let's place it at ```src/utils/isOverlapRecurrentAppointment.js```

**2) Let's simplify the component code:**
```suggestion
  handleAppointmentActions(event, newAppointment) {
      const recurrentAppointments = defaultData.filter((appointment) => appointment?.recurrenceRule)
          .map((appointment) => ({
            ...appointment,
            startDate: new Date(appointment.startDate),
            endDate: new Date(appointment.endDate),
          }));

      for (const recurrentAppointment of recurrentAppointments) {
        const isOverlap = isOverlapRecurrentAppointment(event, recurrentAppointment, newAppointment);

        if (isOverlap) {
          event.cancel = true;
          this.popupVisible = true;
        }
      }
    },
```